tests/test_trajectory.py

```diff
@@ -0,0 +1,305 @@
+    assert trajectory_from_deviation_survey.measured_depths[0] == 1001.0
+
+
+def test_load_from_wellspec(example_model_and_crs):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    grid = RegularGrid(model,
+                       extent_kji = (4, 4, 3),
+                       dxyz = (50.0, -50.0, 50.0),
+                       origin = (0.0, 0.0, 100.0),
+                       crs_uuid = crs.uuid,
+                       set_points_cached = True)
+    grid.write_hdf5()
+    grid.create_xml(write_geometry = True)
+    wellspec_file = os.path.join(model.epc_directory, 'wellspec.dat')
+    well_name = 'DOGLEG'
+    source_df = pd.DataFrame([[2, 2, 1, 0.0, 0.0, 0.0, 0.25], [2, 2, 2, 0.45, -90.0, 2.5, 0.25],
+                              [2, 3, 2, 0.45, -90.0, 1.0, 0.20], [2, 3, 3, 0.0, 0.0, -0.5, 0.20]],
+                             columns = ['IW', 'JW', 'L', 'ANGLV', 'ANGLA', 'SKIN', 'RADW'])
+    with open(wellspec_file, 'w') as fp:
+        fp.write(F'WELLSPEC {well_name}\n')
+        for col in source_df.columns:
+            fp.write(f' {col:>6s}')
+        fp.write('\n')
+        for row_index in range(len(source_df)):
+            row = source_df.iloc[row_index]
+            for col in source_df.columns:
+                if col in ['IW', 'JW', 'L']:
+                    fp.write(f' {int(row[col]):6d}')
+                else:
+                    fp.write(f' {row[col]:6.2f}')
+            fp.write('\n')
+
+    # --------- Act ----------
+    # Create a trajectory from the WELLSPEC file
+    trajectory_from_wellspec = resqpy.well.Trajectory(parent_model = model,
+                                                      well_name = well_name,
+                                                      length_uom = 'm',
+                                                      grid = grid,
+                                                      wellspec_file = wellspec_file)
+
+    # --------- Assert ----------
+    assert trajectory_from_wellspec is not None
+    assert trajectory_from_wellspec.knot_count == len(trajectory_from_wellspec.control_points)
+
+
+def test_load_from_ascii_file(example_model_and_crs):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    well_names = [None, 'Coconut']
+    elevation = 100
+    # Create a measured depth datum
+    datum = resqpy.well.MdDatum(parent_model = model,
+                                crs_uuid = crs.uuid,
+                                location = (0, 0, -elevation),
+                                md_reference = 'kelly bushing')
+    mds = np.array([300, 310, 330, 340])
+    zs = mds - elevation
+    source_dataframe = pd.DataFrame({
+        'MD': mds,
+        'X': [1, 2, 3, 4],
+        'Y': [1, 2, 3, 4],
+        'Z': zs,
+        'WELL': ['Coconut', 'Coconut', 'Mango', 'Mango']
+    })
+    deviation_survey_file_path = os.path.join(model.epc_directory, 'deviation_survey.csv')
```

Comment:
strictly speaking this is trajectory data rather than a deviation survey (which measures inclinations and azimuths); in the industry the derived trajectory is often lazily referred to as a deviation survey but as RESQML makes a clear distinction, it might help if we stick with RESQML compatible variable names

Reply:
I didn't appreciate the difference before you pointed it out! As RESQML does make a clear distinction, I've added a TODO statement to change the argument name in the Trajectory_init.